cirq/sim/mux.py

```diff
@@ -23,11 +23,19 @@
 
 from cirq import circuits, protocols, study, devices, ops, value
 from cirq.sim import sparse_simulator, density_matrix_simulator
+from cirq.sim.clifford import clifford_simulator
 
 if TYPE_CHECKING:
     import cirq
 
 
+def _is_clifford_circuit(program: 'cirq.Circuit') -> bool:
+    return all(
+        op.gate in clifford_simulator.CliffordSimulator.get_supported_gates()
```

Comment:
This check will fail on gates that decompose into Clifford operations, such as cirq.ISWAP.

I think what we need to do here is to define an is clifford protocol as well as formalize the `decompose_into_clifford` protocol. This line would then reduce to `has_clifford(circuit)`.

Reply:
Added a TODO for now